### PR TITLE
support partition models which are partitioned on the primary key

### PR DESCRIPTION
**Pull request by snopoke**

### CODE REVIEW

File: corehq/apps/dump_reload/sql/load.py [hunk 1]
```diff
@@ -160,11 +160,15 @@ def _group_objects_by_db(objects):
...
-                    if field.column == model.partition_attr
-                ][0]
-                partition_value = obj['fields'][field.name]
+                try:
+                    field = [
+                        field for field in model._meta.fields
+                        if field.column == model.partition_attr
+                    ][0]
+                    partition_value = obj['fields'][field.name]
+                except KeyError:
```
Comment (millerdev) [inline corehq/apps/dump_reload/sql/load.py:169]: Is the last line the only one that is expected to raise `KeyError`? Would this work?
```suggestion
                field = [
                    field for field in model._meta.fields
                    if field.column == model.partition_attr
                ][0]
                try:
                    partition_value = obj['fields'][field.name]
                except KeyError:
```

File: corehq/apps/dump_reload/sql/load.py [hunk 2]
```diff
@@ -160,11 +160,15 @@ def _group_objects_by_db(objects):
...
+                    if model.partition_attr == model._meta.pk.attname:
+                        partition_value = obj['pk']
```
Comment (millerdev) [inline corehq/apps/dump_reload/sql/load.py:171]: `partition_value` will not be set if the condition evaluates to false, and I think that would either cause a `NameError` on the next line or use `partition_value` from the previous loop iteration which seems wrong.

**Review by millerdev — COMMENTED** [review]
The code looks like it has a bug 🐞

File: corehq/apps/dump_reload/sql/load.py (hunk 2 above)
Comment (snopoke) [reply corehq/apps/dump_reload/sql/load.py:171]: yea, I guess that should raise an exception

Comment (snopoke) [reply corehq/apps/dump_reload/sql/load.py:172]: 8c695c42a7f

**Review by millerdev — APPROVED** [review]

**Pull request merged**